docs/make.jl

```diff
@@ -0,0 +1,62 @@
+pushfirst!(LOAD_PATH, joinpath(@__DIR__, "..")) # add Oceanostics environment to docs
+using Pkg
+CI = get(ENV, "CI", nothing) == "true" || get(ENV, "GITHUB_TOKEN", nothing) !== nothing
+CI && Pkg.instantiate()
```

Comment:
is this needed?

Reply:
The `CI` boolean is used here https://github.com/tomchor/Oceanostics.jl/blob/574bb78cc36106b25440c2eb64ce56826435dd1b/docs/make.jl#L38-L42

to make links work locally. Otherwise the browser can't resolve them. It's suggested [here](https://documenter.juliadocs.org/stable/man/guide/#Building-an-Empty-Document).

Reply:
ok, so CI is just a boolean? 

I never had trouble with links locally. Perhaps I'm doing it without knowing? as far as I know I usually do 
`prettyurls = get(ENV, "CI", nothing) == "true"`

Where exactly in the link you pointed out is this discussed! I'd like to use that in other repos :)

Reply:
Btw, I was mostly thinking about the `Pkg.instantiate()`. I think there is some confusion here. Since the github action .yml file uses `Pkg.develop()` then there shouldn't be any need to instantiate....

Reply:
Also the line `pushfirst!(LOAD_PATH, joinpath(@__DIR__, ".."))` is not needed if we use `Pkg.develop()` as far as I know. 

I think there is mixed thinking here: part from using `Pkg.develop()` and partly from imposing a fixed set of package/deps versions via a `docs/Manifest.toml`.

(And there is definitely I am also partly confused on this btw...)

Reply:
> ok, so CI is just a boolean?
> 
> I never had trouble with links locally. Perhaps I'm doing it without knowing? as far as I know I usually do `prettyurls = get(ENV, "CI", nothing) == "true"`
> 
> Where exactly in the link you pointed out is this discussed! I'd like to use that in other repos :)

It's in the third note in that section!

> Btw, I was mostly thinking about the `Pkg.instantiate()`. I think there is some confusion here. Since the github action .yml file uses `Pkg.develop(); Pkg.instantiate()` then here shouldn't be any need to instantiate again....

Ah, I see. I think we need that line there when building the docs locally. At least things didn't work for me before adding that. But maybe as you said starting the Julia env with `develop` might help? I'm not too familiar with the `develop` option...

Reply:
If we'll use `Pkg.develop()` then there is no need to add a `docs/Manifest.toml`. I believe as it stands now there are two instantiations and double work been happening installing dependencies etc...

Were you trying to build the docs locally using the same command as in the docs GitHub action? That should have worked.

Reply:
> Also the line `pushfirst!(LOAD_PATH, joinpath(@__DIR__, ".."))` is not needed if we use `Pkg.develop()` as far as I know.
> 
> I think there is mixed thinking here: part from using `Pkg.develop()` and partly from imposing a fixed set of package/deps versions via a `docs/Manifest.toml`.
> 
> (And there is definitely I am also partly confused on this btw...)

I'll try to remove the Manifest see what happens. I thought including it was common practice but that might not be case.

Reply:
> > ok, so CI is just a boolean?
> > I never had trouble with links locally. Perhaps I'm doing it without knowing? as far as I know I usually do `prettyurls = get(ENV, "CI", nothing) == "true"`
> > Where exactly in the link you pointed out is this discussed! I'd like to use that in other repos :)
> 
> It's in the third note in that section!

Hm... It suggests `prettyurls = get(ENV, "CI", nothing) == "true"`. So what does `|| get(ENV, "GITHUB_TOKEN", nothing) !== nothing` do?

Reply:
> > Also the line `pushfirst!(LOAD_PATH, joinpath(@__DIR__, ".."))` is not needed if we use `Pkg.develop()` as far as I know.
> > I think there is mixed thinking here: part from using `Pkg.develop()` and partly from imposing a fixed set of package/deps versions via a `docs/Manifest.toml`.
> > (And there is definitely I am also partly confused on this btw...)
> 
> I'll try to remove the Manifest see what happens. I thought including it was common practice but that might not be case.

No it's not common practice. It can be done either way. E.g.,
- Oceananigans.jl include a `docs/Manifest.toml` but then need to use:
  ```Julia
  julia --project -e 'using Pkg; Pkg.instantiate()' # to instantiate Oceananigans+deps
  julia --project=docs/ -e 'using Pkg; Pkg.instantiate()' # to instantiate the docs project+deps
  julia --project=docs/ docs/make.jl # to build the docs
  ```
  but _also_ at the top of `make.jl` file we need to add `pushfirst!(LOAD_PATH, joinpath(@__DIR__, ".."))`

- GeophysicalFlows.jl _don't_ include a `docs/Manifest.toml` and then use
  ```Julia
  julia --project=docs/ -e 'using Pkg; Pkg.develop(PackageSpec(path=pwd())); Pkg.instantiate()'
  ```
  to build the docs. The `make.jl` file does _not_ need to do anything `pushfirst!(LOAD_PATH, joinpath(@__DIR__, ".."))` since the `Pkg.develop()` brings the current version of the package into the `docs` project.

What this PR does is a hybrid of the two.

Reply:
> > > ok, so CI is just a boolean?
> > > I never had trouble with links locally. Perhaps I'm doing it without knowing? as far as I know I usually do `prettyurls = get(ENV, "CI", nothing) == "true"`
> > > Where exactly in the link you pointed out is this discussed! I'd like to use that in other repos :)
> > 
> > 
> > It's in the third note in that section!
> 
> Hm... It suggests `prettyurls = get(ENV, "CI", nothing) == "true"`. So what does `|| get(ENV, "GITHUB_TOKEN", nothing) !== nothing` do?

I started off with a `make.jl` file in one of [these examples](https://documenter.juliadocs.org/stable/man/examples/) and that line came with it. But you're right that I probably shouldn't include the stuff that doesn't have an obvious function. I'll change that line to what's suggested by the docs.